ruby/private/bundle/create_bundle_build_file.rb

```diff
@@ -33,30 +33,33 @@
     name = "{name}",
     srcs = glob(
       include = [
-        "lib/ruby/{ruby_version}/gems/{name}-{version}/**/*",
-        "bin/*"
+        ".bundle/config",
+        "{gem_lib_files}",
+        {gem_binaries}
       ],
       exclude = {exclude},
     ),
     deps = {deps},
     includes = ["lib/ruby/{ruby_version}/gems/{name}-{version}/lib"],
-    rubyopt = ["{bundler_setup}"],
   )
 GEM_TEMPLATE
 
 ALL_GEMS = <<~ALL_GEMS
   ruby_library(
     name = "gems",
-    srcs = glob(
-      {gems_lib_files},
-    ),
+    srcs = glob([{gems_lib_files}]) + glob(["bin/*"]),
     includes = {gems_lib_paths},
-    rubyopt = ["{bundler_setup}"],
+  )
+
+  ruby_library(
```

Comment:
Maybe we should split out each binary into its own rule. Given the code below knows which binary belongs to which gem, then we could include the dependency on the gem into the binary rule. This would allow someone to depend on the binary without having to also include the gem that installed it.

e.g.

`deps = ["@bundle//:rubycop.binary"]` instead of  `deps = ["@bundle//:rubycop.binary", "@bundle//:rubycop"]`

Reply:
Or does this work in reverse, where `deps = ["@bundle//:rubycop"]` includes the binary it built as well.

Reply:
We could split each binary into it's own rule, and I was thinking we would generate  BUILD file inside the `bundle/bin` folder, so that they can be referenced as `@bundle//bin:rubocop`